circuit-lib/circuit-lib.circom/src/elgamal-babyjubjub/encrypt.circom

```diff
@@ -0,0 +1,104 @@
+pragma circom 2.1.2;
+
+include "../../node_modules/circomlib/circuits/babyjub.circom";
+include "../../node_modules/circomlib/circuits/bitify.circom";
+include "../../node_modules/circomlib/circuits/escalarmulany.circom";
+include "../../node_modules/circomlib/circuits/escalarmulfix.circom";
+
+template Encode() {
+    signal input plaintext;
+    signal output out[2];
+
+    // baby jubjub curve base point
+    var base[2] = [
+        5299619240641551281634865583518297030282874472190772894086521144482721001553,
+        16950150798460657717958625567821834550301663161624707787222815936182638968203
+    ]; 
+
+    component plaintextBits = Num2Bits(32);
+    component BaseMultiply = EscalarMulFix(32, base);
+
+    var i;
+    plaintext ==> plaintextBits.in;
+    for  (i=0; i<32; i++) {
+        plaintextBits.out[i] ==> BaseMultiply.e[i];
+    }
+
+    BaseMultiply.out ==> out;
+}
+
+template Encrypt() {
+
+    // message encoded as a point on the curve: message = [plaintext].G, G: curve base point
+    signal input message[2];    
+    // secret key nonce          
+    signal input nonceKey;    
+    // public key generated by the receiver             
+    signal input publicKey[2];             
+
+    // ephemeral key: [nonce].Base
+    signal output ephemeralKey[2];  
+    // encrypted message: encryptedMessage = message + [nonceKey].publicKey           
+    signal output encryptedMessage[2];            
+    
+    component isz = IsZero();
+    isz.in <== publicKey[0];
+    
+    component ise = IsEqual();
+    ise.in[0] <== publicKey[1];
+    ise.in[1] <== 1;
+
+    // protect against invalid curve attacks => Public Key shouldn't be the identity point
+    isz.out + ise.out === 0;  
+
+
+    // check the public key is point on curve
+    component  pubKeyIsOnCurve = BabyCheck();
+    pubKeyIsOnCurve.x <== publicKey[0];
+    pubKeyIsOnCurve.y <== publicKey[1];
+ 
+    // check the message is a point on curve
+    component  messageIsOnCurve = BabyCheck();            
+    messageIsOnCurve.x <== message[0];
+    messageIsOnCurve.y <== message[1];
+    
+    // baby jubjub curve base point
+    var base[2] = [
+        5299619240641551281634865583518297030282874472190772894086521144482721001553,
+        16950150798460657717958625567821834550301663161624707787222815936182638968203
+    ];        
+
+    // calculate the ephemeral key
+    component baseMultiply = EscalarMulFix(253, base);
+    
+    // calculate the masking key => [nonce].PublicKey
+    component pubKeyMultiply = EscalarMulAny(253);
+    pubKeyMultiply.p[0] <== publicKey[0];
+    pubKeyMultiply.p[1] <== publicKey[1];
```

Comment:
Naming consisteny - you are mixing pubkey and publicKey please settle for one. We are using publicKey primarily in zk.js. (Only shielded publickey is call pubkey rn and I want to get rid of this naming.)
Please switch to publicKey :)

Reply:
roger that!

Reply:
done!